keys/signer_ed25519_test.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2020 Coinbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package keys
+
+import (
+	"encoding/hex"
+	"testing"
+
+	"github.com/coinbase/rosetta-sdk-go/types"
+	"github.com/stretchr/testify/assert"
+)
+
+var signerEd25519 Signer
+
+func init() {
+	keypair, _ := GenerateKeypair(types.Edwards25519)
+	signerEd25519 = Signer(SignerEd25519{*keypair})
+}
+
+func TestSignEd25519(t *testing.T) {
+	signatureType := types.EcdsaRecovery
```

Comment:
This is not the signature type we want here 😏 , seems like we need some sort of extra validation if this isn't error-ing since an Ed25519 signer can't produce this type of signature